backend/courses/views.py

```diff
@@ -246,13 +263,22 @@ def get_queryset(self):
                 ),
             )
         )
+        check_semester = self.request.GET.get("check_semester")
```

Comment:
1. What's the use case for check_semester? 
2. if there is a strong one, is there anything we could rename to that better captures what it does? (`check_offered_in` or something similar?)

Also, nit: can't we use `self.request.query_params.get(...)`?

Reply:
This is so that PCR URLs can now contain the semester of the course (since course codes may change) - if someone specifies a semester in the URL, they are saying "I want all review data for the topic with this course code in this semester", which is guaranteed to be unique. Then, we only want to fetch current data for the most recent course code in the topic, but only if it is was offered in that semester... otherwise the currently offered course is not in the same topic.